### PR TITLE
fix ptp labels to skip virtual machines

### PR DESCRIPTION
**Pull request by yuvalk**
this is currently done by manually pre-labeling those with 'virtual-machine'

Signed-off-by: Yuval Kashtan <yuvalkashtan@gmail.com>

in our env this will be done by cluster deployment

note: I tried to use ironic introspection data, but that is included only for workers ...

### CODE REVIEW

**Comment by yanirq** [comment]
@yuvalk  is this visible enough for others to know that they need to manual label a machine with `virtualmachine` ?   (or at least consider it)    maybe worth changing to a variable that defaults to virtual machine but you can change it (exporting etc..) ?

**Comment by yuvalk** [comment]
> @yuvalk is this visible enough for others to know that they need to manual label a machine with `virtualmachine` ? (or at least consider it) maybe worth changing to a variable that defaults to virtual machine but you can change it (exporting etc..) ?

I'll add a variable
maybe also worth mentioning in the readme

File: hack/setup-test-cluster.sh [hunk 1]
```diff
@@ -5,6 +5,8 @@ set -e
 # expect oc to be in PATH by default
 export OC_TOOL="${OC_TOOL:-oc}"
 
+export NON_PTP_LABEL="node-role.kubernetes.io/virtual"
```
Comment (yanirq) [inline hack/setup-test-cluster.sh]: do you want to use a default value perhaps like in OC_TOOL above ?  e.g. `"${NON_PTP_LABEL:-node-role.kubernetes.io/virtual}"`  not sure about the syntax with special chars, you'll have to check me on that.

Comment (yuvalk) [reply hack/setup-test-cluster.sh]: yes, of course, you are right, that was the whole point, silly me..

Comment (yuvalk) [reply hack/setup-test-cluster.sh]: tnx for catching this

**Comment by yuvalk** [comment]
@SchSeba - tnx, was going to ask you to take a look

File: hack/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# hack GuideLines
+
+This document describes how you can use the scripts from [`hack`](.) directory
+and gives a bried introduction and explanation of these scripts.
+
+## Overview
+
+The [`hack`](.) directory contains many scripts that ensure continuous development of cnf features,
+enhance the robustness of the code, improve developmetn efficiency, etc.
```
Comment (slintes) [inline hack/README.md]: development

File: hack/README.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+The explanations and descriptions of these sciprts are helpful for contributors.
```
Comment (slintes) [inline hack/README.md]: scripts

File: hack/README.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+## setup-test-cluster.sh
+This script does all the CNF installation parts that are expected to be done by users, beside installation related ones. like labeling nodes and creating needed MachineConfiguration object, etc.
```
Comment (slintes) [inline hack/README.md]: This is a bit confusing IMHO: "it does all the installation parts beside installation".
What about: 
This script does all the cluster setup parts that are expected to be done by admins, like labelling nodes and creating the MachineConfigPool resource, etc.

File: hack/README.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+for PTP it is possible to label nodes as non ptp capable either by
```
Comment (slintes) [inline hack/README.md]: nit: uppercase "For"

File: hack/README.md (hunk 4 above)
Comment (yuvalk) [reply hack/README.md]: I wanted to convey that it does not install CNF just do the pre/post setup
still not sure this is good enough

Comment (yuvalk) [reply hack/README.md]: maybe use your phrasing and add after the dot. "It does not install CNF itself"

**Comment by yuvalk** [comment]
> Good idea to add a README +1
> I have some comments about it though

tnx for reading,
fixed.

File: hack/setup-test-cluster.sh [hunk 6]
```diff
@@ -15,11 +17,11 @@ do
 done
 
 echo "[INFO]: Labeling first node as the ptp grandmaster"
-node=$(${OC_TOOL} get nodes -o name | sed -n 1p)
+node=$(${OC_TOOL} get nodes -o name --selector '!${NON_PTP_LABEL}' | sed -n 1p)
```
Comment (slintes) [inline hack/setup-test-cluster.sh]: `'!${NON_PTP_LABEL}'` -> the single quotes prevent using the actual value, please use double quotes. See failure in e2e test.

File: hack/setup-test-cluster.sh [hunk 7]
```diff
@@ -15,11 +17,11 @@ do
...
 ${OC_TOOL} label --overwrite $node ptp/grandmaster=""
 
 echo "[INFO]: Labeling all the other nodes as ptp slaves"
-nodes=$(${OC_TOOL} get nodes -o name | sed 1d)
+nodes=$(${OC_TOOL} get nodes -o name --selector '!${NON_PTP_LABEL}' | sed 1d)
```
Comment (slintes) [inline hack/setup-test-cluster.sh]: same here

**Comment by yuvalk** [comment]
> @yuvalk: only [openshift-kni](https://github.com/orgs/openshift-kni/people) org members may request cherry picks. You can still do the cherry-pick manually.
> 
> In response to [this](https://github.com/openshift-kni/cnf-features-deploy/pull/122#issuecomment-602796335):
> 
> > /cherrypick release-4.4
> 
> Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md). If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.

@slintes - that shouldnt have happened, now that I am a member. can you check?

**Comment by yuvalk** [comment]
maybe it take some time for the cherrypick robot to pick new members up or something?

**Comment by slintes** [comment]
> maybe it take some time for the cherrypick robot to pick new members up or something?

you need to make your membership public on https://github.com/orgs/openshift-kni/people

**Pull request merged**